tests/fixtures/drupal/modules/phpstan_fixtures/src/EntityQueryWithAccessRule.php

```diff
@@ -22,4 +22,34 @@ public function bar(): void
             ->condition('field_test', 'foo', '=')
             ->execute();
     }
+
+    public function count(): void
+    {
+        \Drupal::entityTypeManager()->getStorage('node')
+            ->getQuery()
+            ->accessCheck(false)
+            ->count()
+            ->execute();
+    }
+
+    public function bug381CountOrderShouldNotMatter(): void
+    {
+        $query = \Drupal::entityTypeManager()->getStorage('node')
+            ->getQuery()
+            ->accessCheck(FALSE)
+            ->count();
+        $count = (int) $query->execute();
+
+        $query = \Drupal::entityTypeManager()->getStorage('node')
+            ->getQuery()
+            ->count();
+        $query->accessCheck(FALSE);
```

Comment:
With `$query->accessCheck(FALSE);` it's ok.

The type of `$query` is not changed after calling `accessCheck` if we don't assign it.
I think we may have the issue elsewhere 🤔

Reply:
Does a type-specifying extension could help? 🤔

Reply:
😬 I'm not sure. This is getting _very_ tricky. I wonder if any other extensions support this kind of wild magic. I reviewed Laravel's and there is some magic in models, but nothing like this

Reply:
Type-specifying won't help here, there are for assert call like

It seems the whole idea with specific types like `EntityQueryType` and co does not help for those cases 🤔 
I checked in phpstan-doctrine to know how they handle the `rowCount` method and other like that and for the moment they only support chaining methods (https://github.com/phpstan/phpstan-doctrine/blob/1.3.x/src/Type/Doctrine/DBAL/QueryBuilder/QueryBuilderExecuteMethodExtension.php)

Reply:
What do you think about doing this in another PR and merge the fix for the other case? I could quickly update the PR to do this

Reply:
I'm all for quick fixes for this

Reply:
@brambaud interesting, for what phpstan-doctrine does, I wonder if we should try that out in instance misses from `($methodName === 'execute')`